app/controllers/tasks_controller.rb

```diff
@@ -20,6 +20,14 @@ def show
     else
       raise ActionController::RoutingError.new('Not Found')
     end
+    @student = @task.student
+    @submissions = @task.submissions #reverse order
+    @submission = @submissions.first
+    if current_user.teacher?
```

Comment:
There have been a small change to a `User` model recently which introduces an `admin` property. Admin can see others task pages and can be a teacher as well.
You should check if a teacher is the on who has this course as well. Check out last changes in `tasks#show`.